Source/LudoEngine/GameEngine.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LudoEngine
+{
+    public class GameEngine
+    {
+
+
+
+
+        private int _numberOfPlayers;
+        public int PiecesPerPlayer { get; set; }
+
+        /// <summary>
+        /// There has to be between 2 and 4 players.
+        /// </summary>
+        public int NumberOfPlayers
+        {
+            get { return _numberOfPlayers; }
+
+            set
+            {
+                if (value < 2)
+                {
+                    throw new IndexOutOfRangeException("There needs to be at least two players.");
```

Comment:
Nice to see that you have inserted some exceptions to implement the requirement of 2-4 players. (Encapsulation)